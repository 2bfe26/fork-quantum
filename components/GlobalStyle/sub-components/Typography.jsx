import { createGlobalStyle } from 'styled-components';

const Typography = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=Nunito+Sans');

  * {
    font-family: 'Nunito Sans', sans-serif;
  }

  body {
    font-size: 16px;
    line-height: 24px;
    letter-spacing: 0.2px;
  }

  h1, h2, h3, h4, h5, h6 {
    margin: 0;
    font-weight: normal;
  }

  h1 {
    line-height: 72px;
    letter-spacing: 0.7px;
    font-size: 48px;
  }

  h2 {
    font-size: 36px;
    line-height: 54px;
    letter-spacing: 0.5px;
  }

  h3 {
    font-size: 28px;
    line-height: 42px;
    letter-spacing: 0.4px;
  }
  
  h4 {
    font-size: 24px;
    line-height: 36px;
    letter-spacing: 0.4px;
  }

  h5 {
    font-size: 20px;
    line-height: 30px;
    letter-spacing: 0.3px;
  }

  h6 {
    font-size: 18px;
    line-height: 27px;
    letter-spacing: 0.3px;
  }
`;

export default Typography;
