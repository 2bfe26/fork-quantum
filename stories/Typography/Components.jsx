import React from 'react';
import styled from 'styled-components';
import LinkTo from '@storybook/addon-links/react';
import { Title } from '@catho-private/quantum-storybook-ui';

import {
  Title as QuantumTitle,
  Subtitle,
  Blockquote,
} from '../../components/GlobalStyle';

// eslint-disable-next-line react/prop-types
const WithNunito = ({ component: Component, ...rest }) => {
  const Nunitted = styled(Component)`
    & {
      font-family: 'Nunito Sans' !important;
    }
  `;
  return <Nunitted {...rest} />;
};

const Font = styled.div`
  * {
    font-family: Montserrat, sans-serif !important;
  }
`;

const Small = styled.small`
  font-size: 10px;
`;

const Tr = styled.tr`
  height: 54px;

  td:first-child {
    width: 140px;
  }

  td:nth-child(2) {
    width: 180px;
  }
`;

const Components = () => (
  <Font>
    <Title>Headings</Title>

    <table>
      <tbody>
        <Tr>
          <td>
            h1 <Small>48px</Small>
          </td>
          <td>
            <code>{'<Title />'}</code>
          </td>
          <td>
            <WithNunito component={QuantumTitle}>Heading One</WithNunito>
          </td>
        </Tr>
        <Tr>
          <td>
            h2 <Small>54px</Small>
          </td>
          <td>
            <code>{'<Title as="h2" />'}</code>
          </td>
          <td>
            <WithNunito component={QuantumTitle} as="h2">
              Heading Two
            </WithNunito>
          </td>
        </Tr>
        <Tr>
          <td>
            h3 <Small>28px</Small>
          </td>
          <td>
            <code>{'<Title as="h3" />'}</code>
          </td>
          <td>
            <WithNunito component={QuantumTitle} as="h3">
              Heading Three
            </WithNunito>
          </td>
        </Tr>
        <Tr>
          <td>
            h4 <Small>24px</Small>
          </td>
          <td>
            <code>{'<Title as="h4" />'}</code>
          </td>
          <td>
            <WithNunito component={QuantumTitle} as="h4">
              Heading Four
            </WithNunito>
          </td>
        </Tr>
        <Tr>
          <td>
            h5 <Small>20px</Small>
          </td>
          <td>
            <code>{'<Title as="h5" />'}</code>
          </td>
          <td>
            <WithNunito component={QuantumTitle} as="h5">
              Heading Five
            </WithNunito>
          </td>
        </Tr>
        <Tr>
          <td>
            h6 <Small>18px</Small>
          </td>
          <td>
            <code>{'<Title as="h6" />'}</code>
          </td>
          <td>
            <WithNunito component={QuantumTitle} as="h6">
              Heading Six
            </WithNunito>
          </td>
        </Tr>
        <Tr>
          <td>
            Subtitle <Small>14px</Small>
          </td>
          <td>
            <code>{'<Subtitle />'}</code>
          </td>
          <td>
            <WithNunito component={Subtitle}>Heading Two</WithNunito>
          </td>
        </Tr>
      </tbody>
    </table>

    <Title>Texts</Title>

    <table>
      <tbody>
        {/* // TODO: waiting spec
        <Tr>
          <td>
            a <Small>inherit</Small>
          </td>
          <td>
            <Link href="/">Lorem ipsum dolor sit amet.</Link>
          </td>
        </Tr>
        <Tr>
          <td>
            p <Small>16px</Small>
          </td>
          <td>
            <P>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur
              lorem magna, molestie at pretium non, consequat sit amet ante.
              Aenean et ultrices elit. Suspendisse quis urna consequat,
              condimentum tortor ut, laoreet magna. Praesent accumsan massa
              pellentesque felis tempus, at semper lacus ullamcorper. Vestibulum
              maximus est est, sed ullamcorper mi pellentesque at. Nunc
              facilisis blandit felis sed sollicitudin. Proin rutrum consectetur
              gravida.
            </P>
          </td>
        </Tr> */}
        <Tr>
          <td>
            blockquote <Small>12px</Small>
          </td>
          <td>
            <code>{'<Blockquote />'}</code>
          </td>
          <td>
            <WithNunito component={Blockquote}>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur
              lorem magna, molestie at pretium non, consequat sit amet ante.
              Aenean et ultrices elit. Suspendisse quis urna consequat,
              condimentum tortor ut, laoreet magna. Praesent accumsan massa
              pellentesque felis tempus, at semper lacus ullamcorper. Vestibulum
              maximus est est, sed ullamcorper mi pellentesque at. Nunc
              facilisis blandit felis sed sollicitudin. Proin rutrum consectetur
              gravida.
            </WithNunito>
          </td>
        </Tr>
      </tbody>
    </table>

    <Title>Lists</Title>

    <table>
      <tbody>
        <Tr>
          <td>
            ul {'>'} li <Small>16px</Small>
          </td>
          <td>
            <LinkTo kind="10. List" story="List">
              {'<List />'}
            </LinkTo>
          </td>
          <td>
            <WithNunito component="ul">
              <li>Lorem ipsum dolor sit amet</li>
              <li>
                Sed do eiusmod tempor incididunt ut labore et dolore magna
                aliqua
              </li>
              <li>Consectetur adipiscing elit</li>
            </WithNunito>
          </td>
        </Tr>
        <Tr>
          <td>
            ol {'>'} li <Small>16px</Small>
          </td>
          <td>
            <LinkTo kind="10. List" story="List">
              {'<List />'}
            </LinkTo>
          </td>
          <td>
            <WithNunito component="ol">
              <li>Lorem ipsum dolor sit amet</li>
              <li>
                Sed do eiusmod tempor incididunt ut labore et dolore magna
                aliqua
              </li>
              <li>Consectetur adipiscing elit</li>
            </WithNunito>
          </td>
        </Tr>
      </tbody>
    </table>
  </Font>
);

export default Components;
