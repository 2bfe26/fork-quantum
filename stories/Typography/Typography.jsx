import React from 'react';
import styled from 'styled-components';
import LinkTo from '@storybook/addon-links/react';
import { HowToImport, Title } from '@catho-private/quantum-storybook-ui';

import Colors from '../../components/Colors/deprecated';

const Font = styled.div`
  font-family: Oxygen;
`;

const H = styled.span`
  font-size: ${props => props.size}px;
  color: ${Colors.SECONDARY['900']};
`;

const Link = styled.a`
  color: ${Colors.PRIMARY['500']};
`;

const P = styled.p`
  font-size: 16px;
`;

const Small = styled.small`
  font-size: 10px;
`;

const Caption = styled.caption`
  padding: 10px;
  text-align: left;
`;

const Tr = styled.tr`
  height: 54px;
`;

const Td = styled.td`
  width: 150px;
`;

const Blockquote = styled.blockquote`
  color: #999;
  font-size: 12px;
  line-height: 18px;
  margin: 0;
`;

const Ul = styled.ul`
  margin: 0;
  padding-left: 18px;
`;

const Ol = styled.ol`
  margin: 0;
  padding-left: 18px;
`;

const Li = styled.li`
  line-height: 24px;
`;

const Typography = () => (
  <Font>
    <p>
      <strong>Oxygen</strong> is the standard typeface for Catho on the web
    </p>

    <p>
      In order to use the default Catho Typography, just import the font css:
    </p>

    <HowToImport from="@cathodevel/quantum/static/typography.css" />

    <p>
      This font also provides all the available{' '}
      <LinkTo kind="1. Foundation" story="Icons">
        Icons
      </LinkTo>
    </p>

    <Title>Headings</Title>

    <table>
      <tbody>
        <Tr>
          <Td>
            h1 <Small>24px</Small>
          </Td>
          <td>
            <H size="24">Heading One</H>
          </td>
        </Tr>
        <Tr>
          <Td>
            h2 <Small>22px</Small>
          </Td>
          <td>
            <H size="22">Heading Two</H>
          </td>
        </Tr>
        <Tr>
          <Td>
            h3 <Small>20px</Small>
          </Td>
          <td>
            <H size="20">Heading Three</H>
          </td>
        </Tr>
        <Tr>
          <Td>
            h4 <Small>18px</Small>
          </Td>
          <td>
            <H size="18">Heading Four</H>
          </td>
        </Tr>
        <Tr>
          <Td>
            h5 <Small>18px</Small>
          </Td>
          <td>
            <H size="18">Heading Five</H>
          </td>
        </Tr>
      </tbody>
    </table>

    <Title>Texts</Title>

    <table>
      <tbody>
        <Tr>
          <Td>
            a <Small>inherit</Small>
          </Td>
          <td>
            <Link href="/">Lorem ipsum dolor sit amet.</Link>
          </td>
        </Tr>
        <Tr>
          <Td>
            p <Small>16px</Small>
          </Td>
          <td>
            <P>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur
              lorem magna, molestie at pretium non, consequat sit amet ante.
              Aenean et ultrices elit. Suspendisse quis urna consequat,
              condimentum tortor ut, laoreet magna. Praesent accumsan massa
              pellentesque felis tempus, at semper lacus ullamcorper. Vestibulum
              maximus est est, sed ullamcorper mi pellentesque at. Nunc
              facilisis blandit felis sed sollicitudin. Proin rutrum consectetur
              gravida.
            </P>
          </td>
        </Tr>
        <Tr>
          <Td>
            blockquote <Small>12px</Small>
          </Td>
          <td>
            <Blockquote>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur
              lorem magna, molestie at pretium non, consequat sit amet ante.
              Aenean et ultrices elit. Suspendisse quis urna consequat,
              condimentum tortor ut, laoreet magna. Praesent accumsan massa
              pellentesque felis tempus, at semper lacus ullamcorper. Vestibulum
              maximus est est, sed ullamcorper mi pellentesque at. Nunc
              facilisis blandit felis sed sollicitudin. Proin rutrum consectetur
              gravida.
            </Blockquote>
          </td>
        </Tr>
      </tbody>
    </table>

    <Title>Lists</Title>

    <table>
      <Caption>Lists have line-height of 24 pixels.</Caption>
      <tbody>
        <Tr>
          <Td>
            ul {'>'} li <Small>16px</Small>
          </Td>
          <td>
            <Ul style={{ marginBottom: '20px' }}>
              <Li>Lorem ipsum dolor sit amet</Li>
              <Li>
                Sed do eiusmod tempor incididunt ut labore et dolore magna
                aliqua
              </Li>
              <Li>Consectetur adipiscing elit</Li>
            </Ul>
          </td>
        </Tr>
        <Tr>
          <Td>
            ol {'>'} li <Small>16px</Small>
          </Td>
          <td>
            <Ol>
              <Li>Lorem ipsum dolor sit amet</Li>
              <Li>
                Sed do eiusmod tempor incididunt ut labore et dolore magna
                aliqua
              </Li>
              <Li>Consectetur adipiscing elit</Li>
            </Ol>
          </td>
        </Tr>
      </tbody>
    </table>
  </Font>
);

export default Typography;
