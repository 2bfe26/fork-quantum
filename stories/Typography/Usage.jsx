import React from 'react';
import styled from 'styled-components';
import Highlight from 'react-highlight';
import LinkTo from '@storybook/addon-links/react';
import { CodeToClipboard } from '@catho-private/quantum-storybook-ui';

const Font = styled.div`
  * {
    font-family: Montserrat, sans-serif !important;
  }
`;

const HighlightWrapper = styled.div`
  position: relative;
`;

const glogalStyleImport = `import { GlobalStyle } from '@catho-private/quantum';`;

const glogalStyleCode = `...
ReactDOM
  .render(() => (
    <>
      <GlobalStyle />
      <App />
    </>
  ), document.getElementById('root'));
`;

const typographyAndIconImport = `import { Typography, IconFont } from '@catho-private/quantum';`;

const Typography = () => (
  <Font>
    <p>
      <strong>Nunito Sans</strong> is the standard typeface for Catho on the web
    </p>

    <p>
      In order to use the default Catho Typography, just import the GlobalStyle
      component:
    </p>

    <HighlightWrapper>
      <Highlight language="bash" className="highlight">
        {glogalStyleImport}
      </Highlight>

      <CodeToClipboard code={glogalStyleImport} />
    </HighlightWrapper>

    <p>Usually, it is used before the application component</p>
    <HighlightWrapper>
      <Highlight language="bash" className="highlight">
        {glogalStyleCode}
      </Highlight>

      <CodeToClipboard code={glogalStyleCode} />
    </HighlightWrapper>

    <p>
      The <code>{'<GlobalStyle />'}</code> will add and set automatically all
      the base configuration for font and{' '}
      <LinkTo kind="1. Foundation" story="Icons">
        Icons
      </LinkTo>
      .
    </p>

    <p>
      If you want to include one of those separately, you can import each one as
      you like to:
    </p>
    <HighlightWrapper>
      <Highlight language="bash" className="highlight">
        {typographyAndIconImport}
      </Highlight>

      <CodeToClipboard code={typographyAndIconImport} />
    </HighlightWrapper>
  </Font>
);

export default Typography;
