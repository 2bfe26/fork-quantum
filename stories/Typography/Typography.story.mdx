import { Meta, Story } from '@storybook/addon-docs';

import { Header } from '../shared'
import { Headings, Texts, Lists } from './Components'

<Meta 
  title="Foundation/Typography" 
  parameters={{
    previewTabs: { 
      'storybook/docs/panel': null, 
      canvas: {
        hidden: true,
      },
    },
  }}
/>

<Header title="Typography">
  Typography - Quantum uses Nunito Sans as the standard typeface to present
  our design and content as clearly and efficiently as possible.
</Header>


## Usage

```js
import { GlobalStyle } from '@catho/quantum';
```

Usually, it is used before the application component

```js
ReactDOM
  .render(() => (
    <>
      <GlobalStyle />
      <App />
    </>
  ), document.getElementById('root'));
```

The `<GlobalStyle />` will add and set automatically all the base configuration for font and [Icons](?path=/docs/foundation--icons).

If you want to include one of those separately, you can import each one as you like to:

```js
import { Typography } from '@catho/quantum';
```

## Components

### Headings

<Headings />

### Texts

<Texts />

### Lists

more info about List component, click [here](?path=/docs/foundation--list).

```js
import { List } from '@catho/quantum';
```

```js
const exampleItemsDefault = [
  { content: 'Lorem ipsum dolor sit amet' },
  { content: 'Sed do eiusmod tempor ' },
  { content: 'Consectetur adipiscing elit' },
];
```

<Lists />