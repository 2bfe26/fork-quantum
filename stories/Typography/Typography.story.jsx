import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';
import { Heading, TabbedView, Tab } from '@catho-private/quantum-storybook-ui';

import Usage from './Usage';
import Components from './Components';

import GlobalStyle from '../../components/GlobalStyle';

const QuantumFont = storyFn => (
  <>
    <GlobalStyle />
    {storyFn()}
  </>
);

const Font = styled.div`
  * {
    font-family: Montserrat, sans-serif !important;
  }

  & > h1:first-child {
    line-height: 1;
    letter-spacing: inherit;
  }

  p strong {
    font-weight: 700;
  }
`;

storiesOf('1. Foundation', module)
  .addDecorator(QuantumFont)
  .add('Typography', () => (
    <Font>
      <Heading title="Typography">
        <TabbedView>
          <Tab title="Usage">
            <Usage />
          </Tab>
          <Tab title="Components">
            <Components />
          </Tab>
        </TabbedView>
      </Heading>
    </Font>
  ));
