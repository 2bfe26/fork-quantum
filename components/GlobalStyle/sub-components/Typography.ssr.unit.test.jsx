/**
 * @jest-environment node
 */

import React from 'react';
import renderer from 'react-test-renderer';
import Typography from './Typography';

describe('Typography component', () => {
  it('Should add standart typeface on SSR via tag link', () => {
    expect(renderer.create(<Typography />).toJSON()).toMatchSnapshot();
  });
});
