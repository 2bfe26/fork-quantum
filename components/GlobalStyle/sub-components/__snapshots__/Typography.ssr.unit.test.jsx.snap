// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography component Should add standart typeface on SSR via tag link 1`] = `
<link
  href="https://fonts.googleapis.com/css?family=Nunito+Sans:400,600,700,400i,600i,700i&display=swap"
  rel="stylesheet"
/>
`;
